### PR TITLE
chore: fail channel loop gracefully

### PR DESCRIPTION
**Pull request by jrandolf**
This PR fails the channel loop when CDP errors as these errors are permanent. The loop will restart from `startListenerFromWindow` in these cases.

### CODE REVIEW

File: src/bidiMapper/domains/script/channelProxy.ts [hunk 1]
```diff
@@ -171,64 +171,64 @@ export class ChannelProxy {
...
             },
           },
-        },
-        realm.browsingContextId
-      );
-    }
+          realm.browsingContextId
+        );
+      }
+    } catch {}
```
Comment (thiagowfx) [inline src/bidiMapper/domains/script/channelProxy.ts:231]: Something is missing here, either some or all of those:

- Rationale for the PR (what problem does it resolve?) --> could just augment the commit message; alternatively (or as well), add a code comment here
- Log the `catch`?
- Why was this comment removed?

```
 // TODO(#294): Remove this loop after the realm is destroyed.
 // Rely on the CDP throwing exception in such a case.
```

**Comment by thiagowfx** [comment]
Adding @Lightning00Blade as I'll be OOO. Don't wait for my  re-review.

File: src/bidiMapper/domains/script/channelProxy.ts [hunk 2]
```diff
@@ -171,63 +174,66 @@ export class ChannelProxy {
...
           },
-        },
-        realm.browsingContextId
-      );
+          realm.browsingContextId
+        );
+      } catch (error) {
+        // If an error is thrown, then the channel is permanently broken, so we
+        // exit the loop.
+        this.#logger?.(LogType.cdp, error);
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/script/channelProxy.ts:234]:
```suggestion
        this.#logger?.(LogType.debug, error);
```

File: src/bidiMapper/domains/script/channelProxy.ts [hunk 3]
```diff
@@ -31,8 +32,9 @@ export class ChannelProxy {
   readonly #properties: Script.ChannelProperties;
 
   readonly #id = uuidv4();
+  readonly #logger?: LoggerFn;
 
-  constructor(channel: Script.ChannelProperties) {
+  constructor(channel: Script.ChannelProperties, logger?: LoggerFn) {
     if (
       ![0, null, undefined].includes(channel.serializationOptions?.maxDomDepth)
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/script/channelProxy.ts:39]: @thiagowfx Side question is this still needed? I thought the deep serialization options landed.

File: src/bidiMapper/domains/script/channelProxy.ts (hunk 1 above)
Comment (jrandolf) [reply src/bidiMapper/domains/script/channelProxy.ts]: Done.

**Review by Lightning00Blade — APPROVED** [review]
LGTM + Is it possible to add a test for this scenario ?

**Comment by jrandolf** [comment]
> Is it possible to add a test for this scenario

Not in a reasonable way since the error is transient.

**Pull request merged**